### PR TITLE
Regenerate the Go client by code-generator

### PR DESCRIPTION
**Pull request by tenzen-y**
Signed-off-by: tenzen-y <yuki.iwai.tz@gmail.com>

**What this PR does / why we need it**:
Regenerate the Go client by code-generator and openapi_generated by openapi-gen. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [x] Clean up Go client (non-breaking change which fixes an issue)

**Release note**:

```release-note
Regenerate the Go client library
```

### CODE REVIEW

File: pkg/apis/serving/v1beta1/inference_service.go [hunk 1]
```diff
@@ -78,6 +78,8 @@ type Batcher struct {
 
 // InferenceService is the Schema for the InferenceServices API
 // +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient
```
Comment (yuzisun) [inline pkg/apis/serving/v1beta1/inference_service.go:82]: Is this required for the client go generation ?

File: pkg/apis/serving/v1alpha1/inference_graph.go [hunk 2]
```diff
@@ -153,22 +169,26 @@ const (
...
-//       - nodeName: breed-classifier
-//         data: $request
-//     breed-classifier:
-//       routerType: Switch
-//       routes:
-//       - service: dog-breed-classifier
-//         condition: { .predictions.class == "dog" }
-//       - service: cat-breed-classifier
-//         condition: { .predictions.class == "cat" }
+//
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/inference_graph.go:178]: Do you use any tool to format these comments?

**Comment by yuzisun** [comment]
@tenzen-y thanks for taking up the work generating the client go code! Mostly lgtm just left a few small questions.

File: pkg/apis/serving/v1beta1/inference_service.go (hunk 1 above)
Comment (yuzisun) [reply pkg/apis/serving/v1beta1/inference_service.go:82]: according to https://kubebuilder.io/migration/legacy/migration_guide_v1tov2.html#migrate-the-apis, we do not need these markers anymore, or it is still needed by client go generator tool?

File: pkg/apis/serving/v1alpha1/inference_graph.go (hunk 2 above)
Comment (tenzen-y) [reply pkg/apis/serving/v1alpha1/inference_graph.go:178]: No, I have used `make fmt`.

File: hack/update-openapigen.sh [hunk 3]
```diff
@@ -38,7 +38,8 @@ test -f $CURRENT_VIOLATION_EXCEPTIONS || touch $CURRENT_VIOLATION_EXCEPTIONS
 # checked-in violation file, prints error message to request developer to
 # fix either the API source code, or the known API rule violation file.
 diff $CURRENT_VIOLATION_EXCEPTIONS $KNOWN_VIOLATION_EXCEPTIONS || \
-    (echo -e "ERROR: \n\t API rule check failed. Reported violations in file $CURRENT_VIOLATION_EXCEPTIONS differ from known violations in file $KNOWN_VIOLATION_EXCEPTIONS. \n"; exit 1)
+    (echo -e "WARNING: \n\t API rule check failed. Reported violations in file $CURRENT_VIOLATION_EXCEPTIONS differ from known violations in file $KNOWN_VIOLATION_EXCEPTIONS. \n")
+#    (echo -e "ERROR: \n\t API rule check failed. Reported violations in file $CURRENT_VIOLATION_EXCEPTIONS differ from known violations in file $KNOWN_VIOLATION_EXCEPTIONS. \n"; exit 1)
```
Comment (tenzen-y) [inline hack/update-openapigen.sh]: @yuzisun 
I found an ~~increase~~ decrese in API violations than `KNOWN_VIOLATION_EXCEPTIONS`.
Would you like to update `KNOWN_VIOLATION_EXCEPTIONS`?

File: pkg/apis/serving/v1beta1/inference_service.go (hunk 1 above)
Comment (tenzen-y) [reply pkg/apis/serving/v1beta1/inference_service.go:82]: Yes, we need to add these markers for code-generator.

File: hack/update-openapigen.sh (hunk 3 above)
Comment (tenzen-y) [reply hack/update-openapigen.sh]: @yuzisun 
I updated the `KNOWN_VIOLATION_EXCEPTIONS`.

**Comment by tenzen-y** [comment]
@yuzisun, Please take another look.

**Comment by yuzisun** [comment]
Thanks @tenzen-y ! Awesome contributions!

/lgtm
/approve

**Comment by yuzisun** [comment]
@tenzen-y I am getting following error when running `make codegen`

```
go env -w GOFLAGS=-mod=mod
/Users/dsun20/github/test/kserve/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths=./pkg/apis/serving/v1alpha1
/Users/dsun20/github/test/kserve/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths=./pkg/apis/serving/v1beta1
hack/update-codegen.sh
chmod: /Users/dsun20/go/pkg/mod/k8s.io/code-generator@v0.23.9/generate-groups.sh: No such file or directory
make: *** [generate] Error 1
```

**Comment by tenzen-y** [comment]
@yuzisun 
Do you have code-gen pkg in GOPATH/pkg/mod/k8s.io/code-generator@${CODEGEN_VERSION}?

https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/update-codegen.sh#L23

**Comment by yuzisun** [comment]
> @yuzisun Do you have code-gen pkg in GOPATH/pkg/mod/k8s.io/code-generator@${CODEGEN_VERSION}?
> 
> https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/update-codegen.sh#L23

No I don't, maybe we should add a step to get this?

**Comment by tenzen-y** [comment]
> > @yuzisun Do you have code-gen pkg in GOPATH/pkg/mod/k8s.io/code-generator@${CODEGEN_VERSION}?
> > https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/update-codegen.sh#L23
> 
> No I don't, maybe we should add a step to get this?

I see. Can you get code-gen with go mod tidy?

**Comment by yuzisun** [comment]
> > > @yuzisun Do you have code-gen pkg in GOPATH/pkg/mod/k8s.io/code-generator@${CODEGEN_VERSION}?
> > > https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/update-codegen.sh#L23
> > 
> > 
> > No I don't, maybe we should add a step to get this?
> 
> I see. Can you get code-gen with go mod tidy?

works now, thanks!  though the `make generate` generates the different violation file.
Screen Shot 2022-10-20 at 11 01 20 PM

**Comment by yuzisun** [comment]
What I generated has relative path while yours has `github.com` prefix, not sure if there is any setting which makes this difference.

**Comment by tenzen-y** [comment]
Does that mean vaiolation_exeptions.list?

https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/violation_exceptions.list

**Comment by yuzisun** [comment]
> Does that mean vaiolation_exeptions.list?
> 
> https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/violation_exceptions.list

that's correct.

**Comment by tenzen-y** [comment]
> > Does that mean vaiolation_exeptions.list?
> > https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/violation_exceptions.list
> 
> that's correct.

Probably, you use older kube-openapi-gen in the following.

https://github.com/kserve/kserve/blob/e126d019ac9598f71ce747c81ad512d0b704f2bf/hack/update-openapigen.sh#L26

Maybe, we need to specify kube-opeapi-gen version same as code-gen.

Can you specify kube-openapi-gen version and run go mod tidy?

**Comment by yuzisun** [comment]
> 

go.mod has `k8s.io/kube-openapi v0.0.0-20220124234850-424119656bbf`, what version I need to match codegen?

**Comment by tenzen-y** [comment]
> > 
> 
> go.mod has `k8s.io/kube-openapi v0.0.0-20220124234850-424119656bbf`, what version I need to match codegen?

@yuzisun Sorry for the late response.
I used `v0.0.0-20220124234850-424119656bbf` k8s.io/kube-openapi to generate that.
Can you change `hack/update-openapigen.sh` in the following and re-generate that with `./hack/hack/update-openapigen.sh`?

- before
https://github.com/kserve/kserve/blob/9499aba3ed4cf2f2d7644a6d08b942b910a33af1/hack/update-openapigen.sh#L25-L29

- after
```shell
 # Generating OpenAPI specification 
 go run k8s.io/kube-openapi/cmd/openapi-gen@v0.0.0-20220124234850-424119656bbf \ 
     --input-dirs ./pkg/apis/serving/v1beta1,./pkg/apis/serving/v1alpha1,knative.dev/pkg/apis,knative.dev/pkg/apis/duck/v1 \ 
     --output-package ./pkg/apis/serving/v1beta1 -o ./ -v 5 --go-header-file hack/boilerplate.go.txt \ 
     -r $CURRENT_VIOLATION_EXCEPTIONS 
```

**Comment by yuzisun** [comment]
> > go.mod has `k8s.io/kube-openapi v0.0.0-20220124234850-424119656bbf`, what version I need to match codegen?
> 
> @yuzisun Sorry for the late response. I used `v0.0.0-20220124234850-424119656bbf` k8s.io/kube-openapi to generate that. Can you change `hack/update-openapigen.sh` in the following and re-generate that with `./hack/hack/update-openapigen.sh`?
> 
> * before
>   https://github.com/kserve/kserve/blob/9499aba3ed4cf2f2d7644a6d08b942b910a33af1/hack/update-openapigen.sh#L25-L29
> * after
> 
> ```shell
>  # Generating OpenAPI specification 
>  go run k8s.io/kube-openapi/cmd/openapi-gen@v0.0.0-20220124234850-424119656bbf \ 
>      --input-dirs ./pkg/apis/serving/v1beta1,./pkg/apis/serving/v1alpha1,knative.dev/pkg/apis,knative.dev/pkg/apis/duck/v1 \ 
>      --output-package ./pkg/apis/serving/v1beta1 -o ./ -v 5 --go-header-file hack/boilerplate.go.txt \ 
>      -r $CURRENT_VIOLATION_EXCEPTIONS 
> ```

Thanks for this, though it still does not work for me and it changed back to relative path when I run make generate

**Comment by tenzen-y** [comment]
> Thanks for this, though it still does not work for me and it changed back to relative path when I run make generate

hmm...

Can you show me the results to re-generate that using the following scripts?

```shell
# Generating OpenAPI specification
OPENAPI_PKG="$GOPATH/pkg/mod/k8s.io/kube-openapi@v0.0.0-20220124234850-424119656bbf"
go run "${OPENAPI_PKG}/cmd/openapi-gen" \
    --input-dirs ./pkg/apis/serving/v1beta1,./pkg/apis/serving/v1alpha1,knative.dev/pkg/apis,knative.dev/pkg/apis/duck/v1 \
    --output-package ./pkg/apis/serving/v1beta1 -o ./ -v 5 --go-header-file hack/boilerplate.go.txt \
    -r $CURRENT_VIOLATION_EXCEPTIONS
```

Also, can you show me the results of the below commands:

```shell
$ go env GOMODCACHE
$ go env GOMOD
```

**Comment by yuzisun** [comment]
Go env GOMOD returns 
/Users/dsun/github/kserve/go.mod

go env GOMODCACHE returns 
/Users/dsun/go/pkg/mod 

after running the above script it basically reverted  everything you generated in your PR and replaced ‘github.com/kserve/kserve/pkg’ to ‘./pkg’

**Comment by tenzen-y** [comment]
> Go env GOMOD returns /Users/dsun/github/kserve/go.mod
> 
> go env GOMODCACHE returns /Users/dsun/go/pkg/mod
> 
> after running the above script it basically reverted everything you generated in your PR and replaced ‘github.com/kserve/kserve/pkg’ to ‘./pkg’

I see. Maybe, that occurred by the directory structure.
I have kserve projects in `$GOPATH/src/github.com/kserve/kserve`, you have one in `$GOPATH/github.com/kserve`.
According to the [go official document](https://go.dev/doc/gopath_code#Command), it would be good to have go projects in `$GOPATH/src/github.com/$USER/$PROJECT`.

Can you move the kserve directory to `/Users/dsun/go/src/github.com/kserve/kserve`?

**Comment by yuzisun** [comment]
> > Go env GOMOD returns /Users/dsun/github/kserve/go.mod
> > go env GOMODCACHE returns /Users/dsun/go/pkg/mod
> > after running the above script it basically reverted everything you generated in your PR and replaced ‘github.com/kserve/kserve/pkg’ to ‘./pkg’
> 
> I see. Maybe, that occurred by the directory structure. I have kserve projects in `$GOPATH/src/github.com/kserve/kserve`, you have one in `$GOPATH/github.com/kserve`. According to the [go official document](https://go.dev/doc/gopath_code#Command), it would be good to have go projects in `$GOPATH/src/github.com/$USER/$PROJECT`.
> 
> Can you move the kserve directory to `/Users/dsun/go/src/github.com/kserve/kserve`?

@tenzen-y This worked! I think we will need to document this on the [developer guide](https://kserve.github.io/website/0.9/developer/developer/), go allows creating project outside of GOPATH so people might get into trouble like me.

**Comment by tenzen-y** [comment]
> > > Go env GOMOD returns /Users/dsun/github/kserve/go.mod
> > > go env GOMODCACHE returns /Users/dsun/go/pkg/mod
> > > after running the above script it basically reverted everything you generated in your PR and replaced ‘github.com/kserve/kserve/pkg’ to ‘./pkg’
> > 
> > 
> > I see. Maybe, that occurred by the directory structure. I have kserve projects in `$GOPATH/src/github.com/kserve/kserve`, you have one in `$GOPATH/github.com/kserve`. According to the [go official document](https://go.dev/doc/gopath_code#Command), it would be good to have go projects in `$GOPATH/src/github.com/$USER/$PROJECT`.
> > Can you move the kserve directory to `/Users/dsun/go/src/github.com/kserve/kserve`?
> 
> @tenzen-y This worked! I think we will need to document this on the [developer guide](https://kserve.github.io/website/0.9/developer/developer/), go allows creating project outside of GOPATH so people might get into trouble like me.

Sounds good.

**Pull request merged**